server/app/services/applicant/ServiceAreaUpdateResolver.java

```diff
@@ -158,15 +178,87 @@ private Boolean doesUpdateContainCorrectedAddress(
 
   private ImmutableList<ServiceAreaInclusion> getExistingServiceAreaInclusionGroup(
       Path serviceAreaPath, ImmutableMap<String, String> updateMap) {
-    String serviceAreaValue = updateMap.get(serviceAreaPath.toString());
-    ImmutableList.Builder<ServiceAreaInclusion> existingServiceAreaInclusionGroupBuilder =
-        ImmutableList.builder();
-    if (serviceAreaValue != null) {
-      existingServiceAreaInclusionGroupBuilder.addAll(
-          ServiceAreaInclusionGroup.deserialize(serviceAreaValue));
+
+    return updateMap.keySet().stream()
+        .filter(x -> x.startsWith(serviceAreaPath.toString()))
+        .map(this::extractIndexFromPath)
+        .filter(x -> x >= 0)
+        .distinct()
+        .sorted()
+        .map(index -> getServiceAreaInclusion(updateMap, serviceAreaPath, index))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  /**
+   * Attempts to find the index for a service_area array.
+   *
+   * @param updateMapKey Key from the updateMap in path form. Example:
+   *     `applicant.question_name.service_area[0].service_area_id`.
+   *     --------------------------------------^
+   * @return the int value of the index or -1 if errors/no matches found
+   */
+  private int extractIndexFromPath(String updateMapKey) {
```

Comment:
Optional: Did you consider parsing this as a path and using the built in arrayIndex() method on that (and does that work? :) If that works I prefer it because there is less custom code, but it's just a preference.

Reply:
Great catch! Completely spaced on that method